commcare_connect/events/models..py

```diff
@@ -0,0 +1,151 @@
+from django.db import models
+
+from commcare_connect.users.models import User
+
+
+class Event(models.Model):
+
+	class Type(models.TextChoices):
+	    INVITE_SENT = "is", gettext("Invite Sent")
+	    RECORDS_APPROVED = "rp", gettext("Records Approved")
+	    RECORDS_FLAGGED = "rf", gettext("Records Flagged")
+	    RECORDS_REJECTED = "rj", gettext("Records Rejected")
+	    PAYMENT_APPROVED = "pp", gettext("Payment Approved")
+	    PAYMENT_ACCRUED = "pa", gettext("Payment Accrued")
+	    PAYMENT_TRANSFERRED = "pf", gettext("Payment Transferred")
+	    NOTIFICATIONS_SENT = "nf", gettext("Notification Sent")
+	    ADDITIONAL_BUDGET_ADDED = "ab", gettext("Additional Budget Added")
+
+    date_created = models.DateTimeField(auto_now_add=True, db_index=True)
+    event_type = models.CharField(max_length='2', choices=Type.choices)
+    user = models.ForeignKey(User, on_delete=models.CASCADE, null=True)
+    opportunity = models.ForeignKey(Opportunity, on_delete=models.PROTECT, null=True)
```

Comment:
is there no way to save additional metadata?

Reply:
For all the metrics so far, I haven't found one which needs more metadata.